db/migrations/20201214160609_update_get_flip_query.sql

```diff
@@ -0,0 +1,70 @@
+-- +goose Up
+
+-- add a new function that works the same as get_flip but takes into account a specific flip_address
+-- this is required because there are now more than one flip contract per ilk
+CREATE OR REPLACE FUNCTION api.get_flip_with_address(bid_id NUMERIC, flip_address TEXT, ilk TEXT, block_height BIGINT DEFAULT api.max_block())
+    RETURNS api.flip_bid_snapshot
+AS
+$$
+WITH ilk_ids AS (SELECT id FROM maker.ilks WHERE ilks.identifier = get_flip_with_address.ilk),
+     address_id AS (SELECT id FROM public.addresses WHERE address = get_flip_with_address.flip_address),
+     kicks AS (SELECT usr
```

Comment:
Good call - I think that we're okay with not including `LIMIT 1` for the ilk and addresses queries because they both have uniqueness constraints on the fields we're querying by (ilk identifier and address). 

I like the idea of failing fast if there is more than one kick - I was trying to get that to work with an `ASSERT` in there, but haven't quite figured out how to get that to work. I'm wondering if folks think that'd be a useful addition?

Reply:
I'm fine with abandoning this line of thought if it's introducing too much complexity. I think in theory we should only have one kick per bid ID per address, but don't think business rules should creep into the schema unless we can back it up with constraints.

Reply:
That's a good point about avoiding business rules in the schema without constraints. Are you thinking that including an `ASSERT` without a constraint on the flip_kick table would be an example of business logic seeping into the schema?

I think you're right that we should only ever have one kick per bid ID per address, since the flipper contract increments the bid id when `kick` is called: https://github.com/makerdao/dss/blob/master/src/flip.sol#L124. But we don't have a constraint for that in the db. 🤔